package/libpiksi/libpiksi/src/settings.c

```diff
@@ -460,23 +460,8 @@ static setting_data_t * setting_create_setting(settings_ctx_t *ctx, const char *
   return setting_data;
 }
 
-static int setting_register(settings_ctx_t *ctx, const char *section,
-                            const char *name, void *var, size_t var_len,
-                            settings_type_t type, settings_notify_fn notify,
-                            void *notify_context, bool readonly)
+static int setting_register(settings_ctx_t *ctx, setting_data_t *setting_data)
```

Comment:
Mostly the interface, setting_register is used by other modules and I wanted to preserve that api such that I didn't need to do shotgun surgery on the rest of buildroot. this keep the changes quarantined to libpiksi. The register wording didn't fit anymore for watched settings since register in the settings context has a specific meaning.